datastore/integration_test.go

```diff
@@ -741,6 +741,7 @@ func TestIntegration_AggregationQueries(t *testing.T) {
 		}
 	}()
 
+	time.Sleep(10 * time.Second)
```

Comment:
Rather than add a sleep here, could we retry the assertions below with a helper function?

The retries could run the query, check the result, if it's no good, try again after waiting 1 second, and do that 10 times. So you'd at worse get an additional 10 second runtime, but probably complete the test sooner.